### ISSUE TITLE
Return `unsafe.Pointer` instead of raw C type

### ISSUE THREAD

**Issue by l0rem1psum** [asticode/go-astiav#53]
As discussed in a previous PR https://github.com/asticode/go-astiav/pull/44, the raw C struct of `AVFrame` is returned instead of a `unsafe.Pointer`.

However, after some fiddling around, I finally discovered the difference of the two approaches.

Currently, the raw C struct returned will have a type of `*astiav._Ctype_struct_AVFrame`, and as the name suggests, this type is unexported. Hence, it comes with all the inconveniences of returning an unexported type.

To make it usable, one unavoidably has to do the following:

![image](https://github.com/asticode/go-astiav/assets/28675546/b47063b3-6073-41ad-82de-473d6f995467)

In conclusion, it's better to just return `unsafe.Pointer` and let the caller do the casting.

**Comment by asticode** [comment]
Thanks for the detailed explanation 👍 The best would be to replace `UnsafeTypedPointer` to `UnsafePointer` which would indeed return an `unsafe.Pointer`. Do you feel like creating a PR for this?
